frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-cpu-patches.ts

```diff
@@ -0,0 +1,23 @@
+import { VMLikeEntityKind } from '../../../types';
+import { PatchBuilder, PatchOperation } from '../../utils/patch';
+import { getVMLikePatches } from '../vm-template';
+import { getCPU, asVM } from '../../../selectors/vm';
+
+export const getDedicatedCpuPatch = (
+  vmLikeEntity: VMLikeEntityKind,
+  dedicatedCpuPlacement: boolean,
+) => {
+  const isCPUAvailable = !!getCPU(asVM(vmLikeEntity));
+
+  const patches = [
+    new PatchBuilder(
+      isCPUAvailable
+        ? '/spec/template/spec/domain/cpu/dedicatedCpuPlacement'
+        : '/spec/template/spec/domain/cpu',
+    )
+      .setOperation(PatchOperation.REPLACE)
+      .setValue(isCPUAvailable ? dedicatedCpuPlacement : { cores: 1, dedicatedCpuPlacement })
```

Comment:
@suomiy in case cpu is not available, do we even want to give an option to patch `dedicatedCpuPlacement`?

Reply:
I don't see a reason why not. We should just ensure there is no overlap between resources.requests.cpu and domain.cpu and have CPUs in just one place.